aws/data_source_aws_storagegateway_local_disk.go

```diff
@@ -0,0 +1,76 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/storagegateway"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsStorageGatewayLocalDisk() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsStorageGatewayLocalDiskRead,
+
+		Schema: map[string]*schema.Schema{
+			"disk_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"disk_path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"gateway_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func dataSourceAwsStorageGatewayLocalDiskRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).storagegatewayconn
+
+	input := &storagegateway.ListLocalDisksInput{
+		GatewayARN: aws.String(d.Get("gateway_arn").(string)),
+	}
+
+	log.Printf("[DEBUG] Reading Storage Gateway Local Disk: %s", input)
+	output, err := conn.ListLocalDisks(input)
+	if err != nil {
+		return fmt.Errorf("error reading Storage Gateway Local Disk: %s", err)
+	}
+
+	if output == nil || len(output.Disks) == 0 {
+		return errors.New("no results found for query, try adjusting your search criteria")
+	}
+
+	diskPath := d.Get("disk_path").(string)
+	var matchingDisks []*storagegateway.Disk
+
+	for _, disk := range output.Disks {
+		if aws.StringValue(disk.DiskPath) == diskPath {
+			matchingDisks = append(matchingDisks, disk)
+		}
+	}
+
+	if len(matchingDisks) == 0 {
+		return errors.New("no results found for query, try adjusting your search criteria")
+	}
+
+	if len(matchingDisks) > 1 {
+		return errors.New("multiple results found for query, try adjusting your search criteria")
+	}
+
+	disk := matchingDisks[0]
```

Comment:
again do we foresee multiple matches being something that comes later?

Reply:
We would create a separate pluralized data source (e.g. `aws_storagegateway_local_disks`) in that case